modules/seqexec/server/src/main/scala/seqexec/server/flamingos2/Flamingos2ControllerEpics.scala

```diff
@@ -41,77 +39,65 @@ object Flamingos2ControllerEpics extends Flamingos2Controller {
-      case LyotWheel.H1         => "Hart1_G5833"
-      case LyotWheel.H2         => "Hart2_G5834"
-    }
-  )
-
-  implicit val encodeGrismPosition: EncodeEpicsValue[Grism, String] = EncodeEpicsValue((a: Grism)
-    => a match {
-      case Grism.Open    => "Open"
-      case Grism.R1200HK => "HK_G5802"
-      case Grism.R1200JH => "JH_G5801"
-      case Grism.R3000   => "R3K_G5803"
-      case Grism.Dark    => "DK_G5804"
-    }
-  )
+  implicit val encodeWindowCoverPosition: EncodeEpicsValue[WindowCover, String] = EncodeEpicsValue {
+    case WindowCover.OPEN  => "Open"
+    case WindowCover.CLOSE => "Closed"
+  }
+
+  implicit val encodeDeckerPosition: EncodeEpicsValue[Decker, String] = EncodeEpicsValue {
+    case Decker.IMAGING   => "Open"
+    case Decker.LONG_SLIT => "Long_Slit"
+    case Decker.MOS       => "Mos"
+  }
+
+  implicit val encodeFPUPosition: EncodeEpicsValue[FocalPlaneUnit, (String, String)] = EncodeEpicsValue {
+    case FocalPlaneUnit.Open        => ("Open", "null")
+    case FocalPlaneUnit.GridSub1Pix => ("sub1-pix_grid", "null")
+    case FocalPlaneUnit.Grid2Pix    => ("2-pix_grid", "null")
+    case FocalPlaneUnit.Slit1Pix    => ("1pix-slit", "null")
+    case FocalPlaneUnit.Slit2Pix    => ("2pix-slit", "null")
+    case FocalPlaneUnit.Slit3Pix    => ("3pix-slit", "null")
+    case FocalPlaneUnit.Slit4Pix    => ("4pix-slit", "null")
+    case FocalPlaneUnit.Slit6Pix    => ("6pix-slit", "null")
+    case FocalPlaneUnit.Slit8Pix    => ("8pix-slit", "null")
+    case FocalPlaneUnit.Custom(s)   => ("null", s)
+  }
+
+  implicit val encodeFilterPosition: EncodeEpicsValue[Filter, String] = EncodeEpicsValue {
+    case Filter.OPEN    => "Open"
+    case Filter.Y       => "Y_G0811"
+    case Filter.F1056   => "F1056"
+    case Filter.F1063   => "F1063"
+    case Filter.J_LOW   => "J-lo_G0801"
+    case Filter.J       => "J_G0802"
+    case Filter.H       => "H_G0803"
+    case Filter.K_LONG  => "K-long_G0812"
+    case Filter.K_SHORT => "Ks_G0804"
+    case Filter.JH      => "JH_G0809"
+    case Filter.HK      => "HK_G0806"
+    case Filter.DARK    => "DK_G0807"
+    case Filter.K_BLUE  => "K-blue_G0814"
+    case Filter.K_RED   => "K-red_G0815"
+  }
+
+  implicit val encodeLyotPosition: EncodeEpicsValue[Lyot, String] = EncodeEpicsValue {
+    case LyotWheel.OPEN       => "f/16_G5830"
+    case LyotWheel.HIGH       => "null"
+    case LyotWheel.LOW        => "null"
+    case LyotWheel.GEMS_OVER  => "GEMS_over_G5836"
+    case LyotWheel.GEMS_UNDER => "GEMS_under_G5835"
+    case LyotWheel.GEMS       => "Gems_G5835"
+    case LyotWheel.H1         => "Hart1_G5833"
+    case LyotWheel.H2         => "Hart2_G5834"
+  }
+
+  implicit val encodeGrismPosition: EncodeEpicsValue[Grism, String] = EncodeEpicsValue {
+    case Grism.Open    => "Open"
+    case Grism.R1200HK => "HK_G5802"
+    case Grism.R1200JH => "JH_G5801"
+    case Grism.R3000   => "R3K_G5803"
+    case Grism.Dark    => "DK_G5804"
+  }
```

Comment:
Idea gracefully pointed that those were easier to write like this.

Reply:
Should we consider making these enums from the db like on gem?